src/main/java/com/openlattice/controllers/OrganizationsController.java

```diff
@@ -234,10 +234,30 @@ public OrganizationIntegrationAccount getOrganizationIntegrationAccount( @PathVa
             produces = MediaType.APPLICATION_JSON_VALUE )
     public Map<UUID, Set<OrganizationEntitySetFlag>> assembleEntitySets(
             @PathVariable( ID ) UUID organizationId,
-            @RequestBody Set<UUID> entitySetIds ) {
+            @RequestBody Map<UUID, Integer> refreshRatesOfEntitySets ) {
         final var authorizedPropertyTypesByEntitySet =
-                getAuthorizedPropertiesForMaterialization( organizationId, entitySetIds );
-        return assembler.materializeEntitySets( organizationId, authorizedPropertyTypesByEntitySet );
+                getAuthorizedPropertiesForMaterialization( organizationId, refreshRatesOfEntitySets.keySet() );
+
+        // convert mins to millisecs
+        final Map<UUID, Long> refreshRatesInMilliSecsOfEntitySets = new HashMap<>( refreshRatesOfEntitySets.size() );
+        refreshRatesOfEntitySets.forEach(
+                ( entitySetId, refreshRateInMins ) -> {
+                    Long value = null;
+                    if ( refreshRateInMins != null ) {
+                        if ( refreshRateInMins < 1 ) {
+                            throw new IllegalArgumentException( "Minimum refresh rate is 1 minute." );
+                        }
+                        value = refreshRateInMins.longValue() * 3600L;
```

Comment:
It might be worth adding a `getRefreshRateMillisFromMins` function that does all the validation and conversion (or return null if null) for consistency in conversion rates + minimums

Reply:
Will do!